qa_eval/steps/sparql.py

```diff
@@ -16,28 +34,66 @@ def get_var_to_values(
     return dict(var_to_values)
 
 
-def get_permutation_indices(list1: list, list2: list) -> list:
-    if len(list1) != len(list2) or Counter(list1) != Counter(list2):
-        return []
-
-    indices = []
-    used = [False] * len(list1)
-
-    for item2 in list2:
-        for i in range(len(list1)):
-            if not used[i] and list1[i] == item2:
-                indices.append(i)
-                used[i] = True
-                break
-
-    return indices
+def parse_dict2table(
+    reference_vars: Union[list[str], tuple[str, ...]],
+    reference_var_to_values: dict[str, list],
+) -> list[str]:
+    result = []
+    num_rows = len(reference_var_to_values[reference_vars[0]])
+    for row_idx in range(num_rows):
+        row = []
+        for reference_var in reference_vars:
+            val = reference_var_to_values[reference_var][row_idx]
+            if isinstance(val, float):
+                val = truncate(val, 5)
+            if isinstance(val, int):
+                print(val)
+                val = float(val)
+                print(str(val))
+            val = str(val)
+            row.append(val)
+        result.append("".join(row))
+    return result
+
+
+def compare_values(
+    reference_vars: list[str],
+    reference_var_to_values: dict[str, list],
+    actual_vars: Union[list[str], tuple[str, ...]],
+    actual_var_to_values: dict[str, list],
+    results_are_ordered: bool,
+) -> bool:
+
+    if len(reference_vars) > len(actual_vars):
+        return False
```

Comment:
We call this function with the required vars only (I'm not sure if this is intentional or not), and we also have this check
```
if len(required_vars) > len(actual_vars):
        return 0.0
```

prior of calling this function. This check here looks like a duplicate. Also, it would be better if we name the local variable here also `required_vars` . If it's named `reference_vars` I'm under the impression that these are all reference vars, including the optional ones